classes/publication/Repository.php

```diff
@@ -148,21 +148,33 @@ public function version(Publication $publication): int
     /** @copydoc \PKP\publication\Repository::setStatusOnPublish() */
     protected function setStatusOnPublish(Publication $publication)
     {
-        // A publication may be scheduled in a future issue. In such cases,
-        // the `datePublished` should remain empty and the status should be set to
-        // scheduled.
-        //
-        // If there is no assigned issue, the journal may be using a continuous
-        // publishing model in which articles are published right away.
+
+        // A publication may be published without an issue. Use current date if no
+        // custom `datePublished` is set. Status is set to published.
+
+        // A publication may be published in a published issue.
+        // In such cases the `datePublished` should be what the user has entered
+        // or fall back to the issue published date. Status should be set to published.
+
+        // A publication may also be scheduled in a future issue. In such cases,
+        // the the status should be set to scheduled and `datePublished`
+        // should remain what it is.
```

Comment:
Can you move this comment to the docblock and make it the following:

```php
/**
 * Set the status and date of publication
 *
 * A publication's status and date of publication will be set depending
 * on the issue it is assigned to. Publications in future issues should
 * be set to `STATUS_SCHEDULED`. Otherwise, they should be set to
 * `STATUS_PUBLISHED` and should inherit the date of publication from the 
 * issue, unless a date of publication has already been set.
 *
 * Usually, the date of publication will be set when the issue is
 * published. In some unusual cases, a publication may be published 
 * without an issue. This may happen in a continuous publishing model 
 * where articles are published right away. In these cases, the date of 
 * publication should be set to the current date, unless a date of 
 * publication has already been set.
 */
```